crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -183,14 +183,14 @@ fn function_definition(allow_self: bool) -> impl NoirParser<NoirFunction> {
         )
 }
 
-/// contract_visibility: 'open' | 'unsafe' | %empty
-fn contract_visibility() -> impl NoirParser<Option<ContractVisibility>> {
-    keyword(Keyword::Open).or(keyword(Keyword::Unsafe)).or_not().map(|token| match token {
-        Some(Token::Keyword(Keyword::Open)) => Some(ContractVisibility::Open),
-        Some(Token::Keyword(Keyword::Unsafe)) => Some(ContractVisibility::Unsafe),
-        None => None,
-        _ => unreachable!("Only open and unsafe keywords are parsed here"),
-    })
+/// contract_visibility: 'open' | 'unconstrained' | %empty
+///
+/// returns (is_unconstrained, is_open) for whether each keyword was present
+fn contract_visibility() -> impl NoirParser<(bool, bool)> {
```

Comment:
I guess I can see why you're retaining the word "visibility" in places - seems you consider it to mean `ContractFunctionType` subtract `ContractFunctionType::Unconstrained`. I wonder if there's a better term that won't be confused with traditional usage. Perhaps "security"/"encryption"?

Reply:
Security, or some variant of may indeed be better. I'll rename to ContractFunctionType or similar for now but we should keep this new name in mind since `open` and friends may have their names/terminology changed later.